3.analysis/scripts/nbconverted/2.gene_set_enrichment_analysis.py

```diff
@@ -63,46 +64,91 @@
+all_GSEA_results['rank'] = (-np.log10(all_GSEA_results.pval))*(all_GSEA_results.es)
+all_GSEA_results.sort_values(by='pval', ascending = True)
+
+neg_GSEA_results['rank'] = (-np.log10(neg_GSEA_results.pval))*(neg_GSEA_results.es)
+neg_GSEA_results.sort_values(by='es', ascending = False)
 
 
 # In[9]:
 
 
-plt.scatter(x=all_GSEA_results['es'],y=all_GSEA_results['pval'].apply(lambda x:-np.log10(x)),s=1)
+plt.figure()
+plt.scatter(x=all_GSEA_results['es'],y=all_GSEA_results['pval'].apply(lambda x:-np.log10(x)),s=10)
+plt.xlabel('log2 Fold Change (ES)')
+plt.ylabel('-log10(pvalue)')
+plt.title('Gene Set Enrichment Analysis')
+
+plt.figure()
+plt.scatter(x=neg_GSEA_results['es'],y=neg_GSEA_results['pval'].apply(lambda x:-np.log10(x)), s=10)
+plt.xlabel('log2 Fold Change (ES)')
+plt.ylabel('-log10(pvalue)')
+plt.title('Control Gene Set Enrichment Analysis')
 
 
 # In[10]:
 
 
-# plot the enrichment results and save to png--this code needs some work done to it in order to work!
-fig = blitz.plot.running_sum(signature, "DIABETIC CARDIOMYOPATHY", library, result=result, compact=False)
-fig.savefig("running_sum.png", bbox_inches='tight')
+# Using VAE generated data
+
+z = 0
+for df in all_signatures:
+    
+    z = z+1
+
+    fig = blitz.plot.running_sum(df, "regulation of transcription from RNA polymerase II promoter in response to hypoxia (GO:0061418)", library, result=result, compact=False)
+    fig.savefig("running_sum_z_" + str(z) + ".png", bbox_inches='tight')
+
+    fig_compact = blitz.plot.running_sum(df, "regulation of transcription from RNA polymerase II promoter in response to hypoxia (GO:0061418)", library, result=result, compact=True)
+    fig_compact.savefig("running_sum_compact_z_" + str(z) + ".png", bbox_inches='tight')
+
+    fig_table = blitz.plot.top_table(df, library, result, n=15)
+    fig_table.savefig("top_table_z_" + str(z) + ".png", bbox_inches='tight')
+
+# Using negative control
+
+z = 0
+for df in negative_control:
+    
+    z = z+1
+
+    fig = blitz.plot.running_sum(df, "regulation of transcription from RNA polymerase II promoter in response to hypoxia (GO:0061418)", library, result=result, compact=False)
+    fig.savefig("running_sum_neg_z_" + str(z) + ".png", bbox_inches='tight')
 
-fig_compact = blitz.plot.running_sum(signature, "PATHOGENIC ESCHERICHIA COLI INFECTION", library, result=result, compact=True)
-fig_compact.savefig("running_sum_compact.png", bbox_inches='tight')
+    fig_compact = blitz.plot.running_sum(df, "regulation of transcription from RNA polymerase II promoter in response to hypoxia (GO:0061418)", library, result=result, compact=True)
+    fig_compact.savefig("running_sum_compact_neg_z_" + str(z) + ".png", bbox_inches='tight')
 
-fig_table = blitz.plot.top_table(signature, library, result, n=15)
-fig_table.savefig("top_table.png", bbox_inches='tight')
+    fig_table = blitz.plot.top_table(df, library, result, n=15)
+    fig_table.savefig("top_table_neg_z_" + str(z) + ".png", bbox_inches='tigh t')
```

Comment:
typo in `tight` - did matplotlib not yell at you for this?

Reply:
I think I accidentally hit the spacebar right before commiting, thanks for catching that!